test/api/v3/integration/tasks/challenges/POST-tasks_challenge_id.test.js

```diff
@@ -0,0 +1,117 @@
+import {
+  generateUser,
+  translate as t,
+} from '../../../../../helpers/api-v3-integration.helper';
+import { v4 as generateUUID } from 'uuid';
+
+describe('POST /tasks/challenge/:challengeId', () => {
+  let user;
+  let guild;
+  let challenge;
+
+  beforeEach(async () => {
+    user = await generateUser({balance: 1});
+    guild = await user.post('/groups', {
+      name: 'Test Public Guild',
+      type: 'guild',
+    });
+    challenge = await user.post('/challenges', {
+      name: 'Test Challenge',
+      shortName: 'TChallenge',
+      leader: user._id,
+      groupId: guild._id,
+    });
+  });
+
+  it('returns error when challenge is not found', async () => {
+    let fakeChallengeId = generateUUID();
+
+    await expect(user.post(`/tasks/challenge/${fakeChallengeId}`, {
+      text: 'test habit',
+      type: 'habit',
+      up: false,
+      down: true,
+      notes: 1976,
+    })).to.eventually.be.rejected.and.eql({
+      code: 404,
+      error: 'NotFound',
+      message: t('challengeNotFound'),
+    });
+  });
+
+  it('returns error when user does not have the challenge', async () => {
+    let userWithoutChallenge = await generateUser();
+
+    await expect(userWithoutChallenge.post(`/tasks/challenge/${challenge._id}`, {
+      text: 'test habit',
+      type: 'habit',
+      up: false,
+      down: true,
+      notes: 1976,
+    })).to.eventually.be.rejected.and.eql({
+      code: 404,
+      error: 'NotFound',
+      message: t('challengeNotFound'),
+    });
+  });
+
+  xit('returns error when non leader tries to edit challenge', async() => {
+    // @TODO needs join chellenge route
```

Comment:
@TheHollidayInn we could just populate the non-leader `challenges` array, or not?